Assets/_Laz/Tests/CheckpointManagerTests.cs

```diff
@@ -0,0 +1,85 @@
+using Laz;
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.SceneManagement;
+using UnityEngine.SceneManagement;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.TestTools;
+
+namespace Tests.LazCharacter
+{
+    public class CheckPointManagerTests : InputTestFixture
+    {
+
+        private LazPlayer _player = new LazPlayer();
+        private CheckPointManager _checkPointManager = null;
+        private LazCoordinatorBehaviour _lazCoordinatorBehaviour = null;
+        private MockLazMovement _mockMovement = new MockLazMovement();
+        private MockLazoProperties _lazoProperties = new MockLazoProperties();
+        private Keyboard _keyboard = null;
+
+        [SetUp]
+        public override void Setup()
+        {
+            base.Setup();
+            _mockMovement = new MockLazMovement();
+            _lazoProperties = new MockLazoProperties();
+            _player = new LazPlayer();
+            EditorSceneManager.LoadSceneAsyncInPlayMode("Assets/_Laz/Scenes/TestingScenes/LazCheckPointManagerTests.unity", new LoadSceneParameters(LoadSceneMode.Single));
+            _keyboard = InputSystem.AddDevice<Keyboard>();
+        }
+
+        [UnityTest]
+        public IEnumerator Test_CheckPointManager_TestInitialize()
+        {
+            for (int i = 0; i < 5; i++)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+
+            // When
+            _checkPointManager = GameObject.FindObjectOfType<CheckPointManager>();
+
+            // Then
+            yield return new WaitForSeconds(0);
+            _checkPointManager.Initialize(_player);
+
+            // Therefore
+            Vector3 _initialPosition = new Vector3(-10f, 0f, 0f);
```

Comment:
I don't think there's actually a way to do this, CheckPointManager caches the position of the initial checkpoint in awake, but it doesn't update as the checkpoint moves. Without messing with script ordering I don't think setting the position in code would have any effect

Reply:
If you think that should be supported I could write the tests anyways and open a bug ticket or something

Reply:
oh yes i see what you mean. We can leave it then

Reply:
I do kind of think that exposing a property on the checkpoint for position and repositioning based on that does seem a little more useful than just caching an initial position, do you think it would be worth making that change?

Reply:
eh not really, useful =/= it's needed.